controllers/users_controller.js

```diff
@@ -20,17 +23,44 @@ const registerNewUser = async (req, res) => {
     });
 
     const savedUser = await user.save();
-    res.json(savedUser);
+    return res.json(savedUser);
   } catch (err) {
     console.warn("Error registering user:", err);
     if (err.code === 11000) {
-      res.status(400).json({ error: "username already exsits."});
+      return res.status(400).json({ error: "username already exsits."});
     } else if (err._message === "User validation failed") {
-      res.status(400).json({ error: "email is not valid. Please enter valid email address"});
+      return res.status(400).json({ error: "email is not valid. Please enter valid email address"});
     } else {
-      res.status(500).json({ error: "An error occurred while registering the user."});
+      return res.status(500).json({ error: "An error occurred while registering the user."});
     }
   }
 };
 
-module.exports = {getAllUsers, registerNewUser};
\ No newline at end of file
+const login = async (req, res) => {
+  try {
+    const {username, email, password} = req.body;
+    const existingUser = await User.findOne({username});
+    const isMatchedpassword = await bcrypt.compare(password, existingUser.password);
+    if (!isMatchedpassword || email != existingUser.email) {
```

Comment:
```suggestion
    if (!isMatchedpassword || email != existingUser?.email) {
```

When the username doesn't exit, `existingUser` will be null.
Consider adding `?` to handle it or really checking whether `existingUser` is defined and returning a more specific error (Decide what's better)